common/types/errors.go

```diff
@@ -101,12 +101,8 @@ func (err RetryTaskV2Error) Error() string {
 	if err.DomainID != nil {
 		printField(sb, "DomainID", *err.DomainID)
 	}
-	if err.WorkflowID != nil {
-		printField(sb, "WorkflowID", *err.WorkflowID)
-	}
-	if err.RunID != nil {
-		printField(sb, "RunID", *err.RunID)
```

Comment:
obviously minor / a passing thought: may be worth putting nil checks into `printField` so you can pass all (top-level) nils straight to it without `if != nil` checks.

Reply:
It would be nicer. 

However `printField` accepts `interface{}` thus we can not simply do `if != nil` inside. It would require some some reflection to check if value within interface is nil. Not sure if this is  worth it.

Reply:
ah, right.  typed nils.
yea, probably not worth it.